tensorflow_addons/layers/tlu.py

```diff
@@ -0,0 +1,116 @@
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Implements Thresholded Linear Unit."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import tensorflow as tf
+
+
+@tf.keras.utils.register_keras_serializable(package='Addons')
+class TLU(tf.keras.layers.Layer):
+    """Thresholded Linear Unit. An activation function which is similar to ReLU
+    but with a learned threshold that benefits models using FRN(Filter Response
+    Normalization). Original paper: https://arxiv.org/pdf/1911.09737.
+
+    Input shape:
+        Arbitrary. Use the keyword argument `input_shape`
+        (tuple of integers, does not include the samples axis)
+        when using this layer as the first layer in a model.
+
+    Output shape:
+        Same shape as the input.
+
+    Arguments:
+        affine: bool. Whether to make it TLU-Affine or not
+        which has the form `max(x, alpha*x + tau)`
+    """
+
+    def __init__(self,
+                 affine=False,
+                 tau_initializer='zeros',
+                 tau_regularizer=None,
+                 tau_constraint=None,
+                 alpha_initializer='zeros',
+                 alpha_regularizer=None,
+                 alpha_constraint=None,
+                 **kwargs):
+        super(TLU, self).__init__(**kwargs)
+        self.supports_masking = True
+        self.affine = affine
+        self.tau_initializer = tf.keras.initializers.get(tau_initializer)
+        self.tau_regularizer = tf.keras.regularizers.get(tau_regularizer)
+        self.tau_constraint = tf.keras.constraints.get(tau_constraint)
+        if self.affine:
+            self.alpha_initializer = tf.keras.initializers.get(
+                alpha_initializer)
+            self.alpha_regularizer = tf.keras.regularizers.get(
+                alpha_regularizer)
+            self.alpha_constraint = tf.keras.constraints.get(alpha_constraint)
+
+    def build(self, input_shape):
+        param_shape = list(input_shape[1:])
+        self.tau = self.add_weight(
+            shape=param_shape,
+            name='tau',
+            initializer=self.tau_initializer,
+            regularizer=self.tau_regularizer,
+            constraint=self.tau_constraint)
+        if self.affine:
+            self.alpha = self.add_weight(
+                shape=param_shape,
+                name='alpha',
+                initializer=self.alpha_initializer,
+                regularizer=self.alpha_regularizer,
+                constraint=self.alpha_constraint)
```

Comment:
It would be better to set `aggregation` and `synchronization` for `add_weight()` as well. This will make it suitable for distributed training out of the box, instead of having to deal with that in the future. [Link](https://www.tensorflow.org/api_docs/python/tf/keras/layers/Layer?version=nightly#add_weight) to docs.

`synchronization=AUTO`, and `aggregation=MEAN` should be ideal.

Reply:
Cool. Will do that.

Reply:
`aggregation=MEAN` Shouldn't we leave this to the user for custom training loops?

Reply:
Nope, it's more related to how you expect the variable to behave. You would only take the mean of these variables, similar to how you would take the mean of the weights of the conv layer, etc. 

I can't think of a situation where you would set this to either `FIRST_REPLICA_ONLY` or `SUM`. But if you do, let me know, and we can reconsider setting this to `MEAN` by default.

Reply:
Hmm agreed